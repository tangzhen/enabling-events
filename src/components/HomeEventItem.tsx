import React from "react";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import FavoriteIcon from "@material-ui/icons/Favorite";
import ShareIcon from "@material-ui/icons/Share";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { ButtonBase } from "@material-ui/core";
import { Link } from "react-router-dom";
import EnablingEvent from "../model/EnablingEvent";

interface HomeEventItemProp {
  event: EnablingEvent;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    title: {
      fontWeight: "bold",
    },
    media: {
      height: 0,
      paddingTop: "56.25%", // 16:9
    },
    expand: {
      marginLeft: "auto",
    },
    cardAction: {
      display: "block",
      textAlign: "initial",
    },
  })
);

export default function HomeEventItem({ event }: HomeEventItemProp) {
  const classes = useStyles();

  return (
    <Card>
      <CardHeader
        action={
          <IconButton aria-label="settings">
            <MoreVertIcon />
          </IconButton>
        }
        title={
          <Typography variant="h6" className={classes.title}>
            {event.title}
          </Typography>
        }
        subheader={event.startDate}
      />
      <ButtonBase
        component={Link}
        to={`/events/${event.id}`}
        className={classes.cardAction}
      >
        <CardMedia
          className={classes.media}
          image={event.bg}
          title={event.title}
        />
        <CardContent>
          <Typography variant="body2" color="textSecondary" component="p">
            {event.summary}
          </Typography>
        </CardContent>
      </ButtonBase>
      <CardActions disableSpacing>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon />
        </IconButton>
        <IconButton aria-label="share">
          <ShareIcon />
        </IconButton>
        <IconButton className={classes.expand} aria-label="show more">
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
    </Card>
  );
}
