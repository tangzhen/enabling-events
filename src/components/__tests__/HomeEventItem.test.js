import React from "react";
import { render } from "../../../test/test-utils";
import HomeEventItem from "../HomeEventItem";

describe("HomeEventItem", () => {
  it("should render component correct", () => {
    const event = {
      id: 1,
      title: "Pancake With Sliced Strawberry",
      bg: "/files/1.jpg",
      summary:
        "A pancake (or hotcake, griddlecake, or flapjack) is a flat cake, often thin and round, prepared from a starch-based batter that may contain eggs, milk and butter.",
      startDate: "2020-10-24T08:52:10+08:00",
      organizer: "Standard Chartered",
      location: "Entrance of National Gallery",
    };
    const { container } = render(<HomeEventItem event={event} />);
    expect(container).toMatchSnapshot();
  });
});
