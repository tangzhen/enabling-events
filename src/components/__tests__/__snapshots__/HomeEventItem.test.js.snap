// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeEventItem should render component correct 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <h6
          class="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
        >
          Pancake With Sliced Strawberry
        </h6>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          20190228
        </span>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-label="settings"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root makeStyles-cardAction-4"
      href="/events/1"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiCardMedia-root makeStyles-media-2"
        style="background-image: url(/files/1.jpg);"
        title="Pancake With Sliced Strawberry"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-content-5 MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          A pancake (or hotcake, griddlecake, or flapjack) is a flat cake, often thin and round, prepared from a starch-based batter that may contain eggs, milk and butter.
        </p>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <div
      class="MuiCardActions-root"
    >
      <button
        aria-label="add to favorites"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="share"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="show more"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-expand-3"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
